drivers/mm/mm_drv_ti_rat.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023 Texas Instruments Incorporated
+ * Copyright (c) 2023 L Lakshmanan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/mm/rat.h>
+#include <zephyr/drivers/mm/system_mm.h>
+#include <zephyr/sys/__assert.h>
+
+address_trans_params translate_config;
+
+static void address_trans_set_region(uint32_t rat_base_addr, uint16_t region_num,
+			uint64_t system_addr, uint32_t local_addr, uint32_t size, uint32_t enable)
+{
+	uint32_t system_addrL, system_addrH;
+
+	if (size > address_trans_region_size_4G) {
+		size = address_trans_region_size_4G;
+	}
+	system_addrL = (uint32_t)(system_addr & ~((uint32_t)(((uint64_t)1 << size) - 1)));
+	system_addrH = (uint32_t)((system_addr >> 32) & 0xFFFF);
+	local_addr = local_addr & ~((uint32_t)(((uint64_t)1 << size) - 1));
+
+	*RAT_CTRL(rat_base_addr, region_num) = 0;
+	*RAT_BASE(rat_base_addr, region_num) = local_addr;
+	*RAT_TRANS_L(rat_base_addr, region_num) = system_addrL;
+	*RAT_TRANS_H(rat_base_addr, region_num) = system_addrH;
+	*RAT_CTRL(rat_base_addr, region_num) = ((enable & 0x1) << 31u) | (size & 0x3F);
+}
+
+static void address_trans_init(address_trans_params *params)
+{
+	uint32_t i;
+
+	if (params != NULL) {
+		translate_config = *params;
+	}
+
+	__ASSERT(translate_config.num_regions < ADDR_TRANSLATE_MAX_REGIONS,
+		 "Exceeding maximum number of regions");
+
+	for (i = 0; i < translate_config.num_regions; i++) {
+		__ASSERT(translate_config.rat_base_addr != 0, "RAT base address cannot be 0");
+		__ASSERT(translate_config.region_config != NULL,
+			 "RAT region config cannot be NULL");
+
+		/* enable regions setup by user */
+		address_trans_set_region(translate_config.rat_base_addr, i,
+					 translate_config.region_config[i].system_addr,
+					 translate_config.region_config[i].local_addr,
+					 translate_config.region_config[i].size, 1);
```

Comment:
can you just pass `translate_config` or `translate_config.region_config`? no need to overload the stack.